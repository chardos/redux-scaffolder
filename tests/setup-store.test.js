const rsk = require('../src/index.js');
const fs = require('fs-extra');
const makeDir = require('make-dir');
const del = require('del');
const tmpPath = `${process.cwd()}/tmp`;

describe('When running rsk in the command line', () => {
  beforeEach(async () => {
    await makeDir('tmp');
    process.argv.splice(2);
  });

  afterEach(async () => {
    await del(`${tmpPath}/*`);
  })

  it('setup-store should create store.js and match the snapshot', async () => {
    process.argv.push('setup-store', '--codeDirectory=tmp');
    await rsk();
    const output = await fs.readFile(`${tmpPath}/store.js`);

    expect(output.toString()).toMatchSnapshot()
  });

  it('setup-store should create reducer/index.js and match the snapshot', async () => {
    process.argv.push('setup-store', '--codeDirectory=tmp');
    await rsk();
    const output = await fs.readFile(`${tmpPath}/reducers/index.js`);

    expect(output.toString()).toMatchSnapshot()
  });
})