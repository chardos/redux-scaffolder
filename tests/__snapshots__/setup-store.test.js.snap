// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When running rsk in the command line When store.js does not already exist setup-store command should create store.js and match the snapshot 1`] = `
"import { createStore, combineReducers } from \\"redux\\";
import reducers from \\"./store\\";

const rootReducer = combineReducers(reducers);

const store = createStore(
  rootReducer,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);

export default store;
"
`;

exports[`When running rsk in the command line When store.js does not already exist setup-store should create reducer/index.js and match the snapshot 1`] = `"export default {}"`;
